### DISCUSSION TITLE
Please someone tell me how I can restart ComfyUI programatically (from the command line or API)

### DISCUSSION THREAD

**Discussion by vesper8** [Comfy-Org/ComfyUI_frontend#1691] [Q&A] [answered]
I've been trying to find a way to do this for 9 months and I just can't figure it out. There is a button inside the ComfyUI manager that lets you restart.. how can I achieve this same functionality programatically?

My issue is that I provision a new cloud instance using SSH to install a bunch of scripts, and then I need to restart ComfyUI once before I can use my new nodes and models.. and I can't figure out how I can do this via SSH or API?

Does anyone have the answer? Many thanks!

**Comment by ltdrdata** [comment, marked as answer]
You can find that implementation from here.

https://github.com/ltdrdata/ComfyUI-Manager/blob/8576f9c97fa2ab50f70a4bd5e6b8bbd18e307cd1/glob/manager_server.py#L1200

https://github.com/ltdrdata/ComfyUI-Manager/blob/8576f9c97fa2ab50f70a4bd5e6b8bbd18e307cd1/js/common.js#L14
